src/Libraries/CoreNodeModelsWpf/Charts/Controls/BarChartControl.xaml.cs

```diff
@@ -0,0 +1,154 @@
+            {
+                if (model.Labels.Count == model.Values.Count && model.Labels.Count > 0)
+                {
+                    var seriesRange = new ColumnSeries[model.Labels.Count];
+
+                    for (var i = 0; i < model.Labels.Count; i++)
+                    {
+                        seriesRange[i] = new ColumnSeries
+                        {
+                            Title = model.Labels[i],
+                            Values = new ChartValues<double>(model.Values[i]),
+                            Fill = model.Colors[i],
+                            Stroke = model.Colors[i],
+                            ColumnPadding = PADDING,
+                            MaxColumnWidth = MAX_COLUMN_WIDTH,
+                        };
+                    }
+
+                    BarChart.Series.AddRange(seriesRange);
+                }
+            }
+        }
+
+        private void NodeModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "DataUpdated")
+            {
+                var model = sender as BarChartNodeModel;
+
+                // Invoke on UI thread
+                this.Dispatcher.Invoke(() =>
+                {
+                    var seriesRange = new ColumnSeries[model.Labels.Count];
+
+                    for (var i = 0; i < model.Labels.Count; i++)
+                    {
+                        seriesRange[i] = new ColumnSeries
+                        {
+                            Title = model.Labels[i],
+                            Values = new ChartValues<double>(model.Values[i]),
+                            Fill = model.Colors[i],
+                            Stroke = model.Colors[i],
+                            ColumnPadding = PADDING,
+                            MaxColumnWidth = MAX_COLUMN_WIDTH,
+                        };
+                    }
+
+                    BarChart.Series.Clear();
+                    BarChart.Series.AddRange(seriesRange);
+                });
+            }
+        }
+
+        private void ThumbResizeThumbOnDragDeltaHandler(object sender, DragDeltaEventArgs e)
+        {
+            var yAdjust = ActualHeight + e.VerticalChange;
+            var xAdjust = ActualWidth + e.HorizontalChange;
+
+            if (this.Parent.GetType() == typeof(Grid))
+            {
+                var inputGrid = this.Parent as Grid;
+
+                if (xAdjust >= inputGrid.MinWidth)
+                {
+                    Width = xAdjust;
+                }
+
+                if (yAdjust >= inputGrid.MinHeight)
+                {
+                    Height = yAdjust;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Unsubscribes from ViewModel events
+        /// </summary>
+        public void Dispose()
+        {
+            this.model.PropertyChanged -= NodeModel_PropertyChanged;
```

Comment:
Is this Dispose function actually called? 

If not, I think you need to do something like this for user controls:

In the contructor:
`Unloaded += Unload;`

And then:
```
private void Unload(object sender, EventArgs e)
{
    this.model.PropertyChanged -= NodeModel_PropertyChanged;
    Unloaded -= Unload;
 }
```